### DISCUSSION TITLE
Access laravel model in resource

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4706] [Help] [answered]
I have a couple of fields in a form (invoice) that all should trigger the same function after a change (`afterStateUpdated`).
In that function I need to pull data from a related model (client) to do some calculations. My kneejerk reaction is: "how can I access the model so I can pull this data".

I have browsed the docs and this forum and I am having a feeling my idea of how to go forward is somewhat false so please forgive me as I still don't get the whole picture.

In my clients I have a couple of fields saving values for hourly, daily and weekly rates. In the invoices those rates should be taken to calculate a final price, e.g. "3 hours time 80 equals 240". So whenever the qty, the unit (hour, day, week) or the rate is changed a new final price will be calculated.

So what I am aiming at is something like `$client = $invoice->client()` and `$client->getRateHour()`.

**Comment by danharrin** [comment, marked as answer]
If its an edit form, then the model is `$record` inside `afterStateUpdated()`
